src/tools/weave/KeyUtils.cpp

```diff
@@ -220,6 +238,53 @@ bool DecodePrivateKey(const uint8_t *keyData, uint32_t keyDataLen, KeyFormat key
     return res;
 }
 
+bool DecodePublicKey(const uint8_t *keyData, uint32_t keyDataLen, KeyFormat keyFormat, const char *keySource, EVP_PKEY *& key)
```

Comment:
There is a lot of similarities between this code and `DecodePrivateKey`, seems like a potential opportunity for refactoring of common code.

Reply:
I think there is some opportunity for refactoring here. Potentially merging `DecodePrivateKey` and `DecodePublicKey` into the same function. This would depend on `DetectKeyFormat` to differentiate between public and private DER formats.

Reply:
Is this something you're willing to tackle as a part of this PR or leaving this as a TODO for the future?  If you're not going to tackle this, who's on the hook to merge this PR?  (you, me or @jaylogue )?

Reply:
I'd like to leave it as a TODO for the future. And most likely near future as we will probably add support for our intermediates and roots to be HSM backed. Can I tackle at that time? 

I don't have access to merge into open weave, so I think either you or @jaylogue can merge.

Reply:
Merged.